src/makeRobot.js

```diff
@@ -38,7 +38,87 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name,
+    version,
+    wheels,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, chip version: ${this.version}, `
+      + `wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(value) {
+      if (value === undefined) {
+        this.coords.y += 1;
+
+        return this;
+      }
+
+      if (value < 0) {
+        return this;
+      }
+      this.coords.y += value;
+
+      return this;
+    },
```

Comment:
it is way simple to do it like that:
```suggestion
    goForward(value = 1) {
      if (value > 0) {
        this.coords.y += 1;
       }
       return this;
    },
```